arduino-ide-extension/src/browser/boards/boards-config-component.tsx

```diff
@@ -0,0 +1,328 @@
+import { DisposableCollection } from '@theia/core/lib/common/disposable';
+import { Event } from '@theia/core/lib/common/event';
+import { FrontendApplicationState } from '@theia/core/lib/common/frontend-application-state';
+import { nls } from '@theia/core/lib/common/nls';
+import * as React from '@theia/core/shared/react';
+import {
+  Board,
+  BoardIdentifier,
+  BoardWithPackage,
+  DetectedPort,
+  findMatchingPortIndex,
+  Port,
+  PortIdentifier,
+} from '../../common/protocol/boards-service';
+import { Defined } from '../../common/types';
+import { NotificationCenter } from '../notification-center';
+import { BoardsConfigDialogState } from './boards-config-dialog';
+import { EditBoardsConfigActionParams } from './boards-service-provider';
+
+namespace BoardsConfigComponent {
+  export interface Props {
+    /**
+     * This is not the real config, it's only living in the dialog. Users can change it without update and can cancel any modifications.
+     */
+    readonly boardsConfig: BoardsConfigDialogState;
+    readonly notificationCenter: NotificationCenter;
+    readonly appState: FrontendApplicationState;
+    readonly onFocusNodeSet: (element: HTMLElement | undefined) => void;
+    readonly onFilteredTextDidChangeEvent: Event<
+      Defined<EditBoardsConfigActionParams['query']>
+    >;
+    readonly onAppStateDidChange: Event<FrontendApplicationState>;
+    readonly onBoardSelected: (board: BoardIdentifier) => void;
+    readonly onPortSelected: (port: PortIdentifier) => void;
+    readonly searchBoards: (query?: {
+      query?: string;
+    }) => Promise<BoardWithPackage[]>;
+    readonly ports: (
+      predicate?: (port: DetectedPort) => boolean
+    ) => readonly DetectedPort[];
+  }
+
+  export interface State {
+    searchResults: Array<BoardWithPackage>;
+    showAllPorts: boolean;
+    query: string;
+  }
+}
+
+export abstract class Item<T> extends React.Component<{
+  item: T;
+  label: string;
+  selected: boolean;
+  onClick: (item: T) => void;
+  missing?: boolean;
+  details?: string;
+}> {
+  override render(): React.ReactNode {
+    const { selected, label, missing, details } = this.props;
+    const classNames = ['item'];
```

Comment:
I see that in other parts of the codebase we are using `classNames from 'classnames';` to deal with adding/removing classes based on props, would it make sense to do the same here rather than a manual `.join`?

Reply:
Good point on the code quality, but I did not change this code but moved it. It was initially in `arduino-ide-extension/src/browser/boards/boards-config.tsx`. See the GitHub diff: https://github.com/arduino/arduino-ide/pull/2165/files#diff-fb37f20bea00881acee3aafddb1ecefcecf41ce59845ca1510da79e918ee0837L50-L75. (You might need to expand the deleted file diff in the web UI.)

I would avoid polishing how IDE2 handles react classes if it's not required for this PR.